### PR TITLE
Document catalog types. Fixes #185.

### LINKED ISSUE #185: Add a list of all catalog types
I think it would be useful to explain the usage of every catalog type, sorted by categories, maybe with a simple usage example.

This is a low-priority issue.

I opened this issue when I found out that the new Command API parses Catalog Types.

List: http://spongepowered.github.io/SpongeAPI/index.html?overview-summary.html

### PR DESCRIPTION
**Pull request by ST-DDT**
Fixes #185.

### CODE REVIEW

File: source/plugin/data/index.rst [hunk 1]
```diff
@@ -36,6 +36,12 @@ On first glance at the API docs, the data API threatens to overwhelm you with lo
 to simply use the data API, you will not have to deal with many of them, as most interfaces found there are just
 specific data manipulators.
 
+CatalogTypes
```
Comment (ST-DDT) [inline source/plugin/data/index.rst:39]: Not sure about the position of this section.

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9d58b85/.

Here are some links to the pages that were modified:

- added: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9d58b85/plugin/data/catalog-types.html
- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/9d58b85/plugin/data/index.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/736#issuecomment-410826678), not to the direct url to the preview._

**Review by ST-DDT — COMMENTED** [review]
I'm not sure where this topic fits bets. I put it under data as a first try.

File: source/plugin/data/catalog-types.rst [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+.. javadoc-import::
+
+    org.spongepowered.api.CatalogType
+    org.spongepowered.api.CatalogTypes
+    org.spongepowered.api.GameRegistry
+    org.spongepowered.api.text.translation.Translatable
+    org.spongepowered.api.util.annotation.CatalogedBy
+
+A big part of the data API consists of special values with a specific meaning. In previous versions of minecraft this
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: "these" (it's plural)

File: source/plugin/data/catalog-types.rst [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+have been numbers such as ``1`` that meant something specific in a given context for example a simple stone in
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: comma needed after "context"

File: source/plugin/data/catalog-types.rst [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+the ``/give`` command. From version to version minecraft moved away from these magics numbers and moved to a more
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: "More recent versions of Minecraft moved away from these magic numbers, and ..."
We should be capitalising Minecraft in general, I believe. (there's a lowercase one in the 2nd sentence too)

File: source/plugin/data/catalog-types.rst [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+declarative way using (namespaced) identifiers such as ``minecraft:stone``. This however brought up two big issues with
+these values:
+
+1. How do I persist them and reload them?
+2. Which values do exist?
+
+Because of this Sponge introduced a new class of data: :javadoc:`CatalogType`\s.
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: comma needed after "this"

File: source/plugin/data/catalog-types.rst [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+``CatalogType``\s provide two methods make them very versatile.
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: "... two methods that make them ..."

File: source/plugin/data/catalog-types.rst [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+First the :javadoc:`CatalogType#getId()` method that provides an easy access to a unique identifier (within that type of
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: "First, the :javadoc:`CatalogType#getId()` method provides an ..."

File: source/plugin/data/catalog-types.rst [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+data) that can be used to save it to a config file and retrieve it back using the 
+:javadoc:`GameRegistry#getType(Class, String)` method. In fact you don't have to do this yourself, you can rely on
+Sponge's :doc:`config serialization <../configuration/serialization>` to do this for you instead. Some ids are not
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: If this is the second method, it probably justifies a second paragraph starting at "Some ids are ..."

File: source/plugin/data/catalog-types.rst [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+really human readable, so there is another method for ``CatalogType``\s that tries to return a human readable name. You
+can also check whether your type is :javadoc:`Translatable`, which gives you a good name to show to your players using
+their locale.
+
+GameRegistry
+============
+
+The :javadoc:`GameRegistry` gives you access to all variants of a ``CatalogType``\s using the
+:javadoc:`GameRegistry#getAllOf(Class)` method, it also offers various other ways of retrieving a ``CatalogType`` and
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: Turn the comma after "method" into a semicolon (or a full stop and start a new sentence, take your pick).

File: source/plugin/data/catalog-types.rst [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+=================
+
+``CatalogType``\s can be easily used in :doc:`../commands/index` using the ``catalogedElement`` which will automatically
+support auto completion for all possible variants of the given type. Read more about it in the
+:doc:`../commands/arguments` section.
+
+Catalog Classes
+===============
+
+Sometimes we just want to reference a ``CatalogType`` entry in our code without actually getting it from the registry,
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: Turn the comma after "registry" into a semicolon.

File: source/plugin/data/catalog-types.rst [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+this is usually the case for default values. To simplify these usecases there are so called catalog classes which
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: "use-cases"
why are they "so called"?

File: source/plugin/data/catalog-types.rst [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+contain static references to types that are either provided by Minecraft itself or Sponge. You can locate these classes
+by following the :javadoc:`CatalogedBy` annotation on your ``CatalogType`` class. If the annotation is missing on that
+class please also check the parent and child classes. If the class has to many variants or variants can be added and
```
Comment (Inscrutable) [inline source/plugin/data/catalog-types.rst]: to -> too

**Review by Inscrutable — CHANGES_REQUESTED** [review]
Excellent, more Spongification! 
I have quibbled with the grammar and punctuation here and there. Whether it works as described is someone else's problem.

**Review by gabizou — APPROVED** [review]
Documentation is good.

**Comment by ST-DDT** [comment]
@worm424 could you please explain why you closed this? If you have done so in a different place, then I haven't found it.

**Comment by Inscrutable** [comment]
<s>I have no idea what happened here.</s>
Too many thumbs on a mobile device. Worm424 sends his apologies ...
Let's be careful out there, OK? 

This PR is probably good to go now.

**Review by Inscrutable — APPROVED** [review]

**Comment by worm424** [comment]
Sorry  @ST-DDT  it won't happen again

**Pull request merged**